### PR TITLE
DOC-7866: PR #103759 - sql: add `sql.log.user_audit.reduced_config.enabled` to support single role checks for role-based audit logs 

### PR DESCRIPTION
**Pull request by florence-crl**
Addresses DOC-7866

Updated Role-based audit logging page, added section for sql.log.user_audit.reduced_config.enabled.

### CODE REVIEW

File: src/current/v23.1/role-based-audit-logging.md [hunk 1]
```diff
@@ -49,6 +51,10 @@ With the [`sql.log.user_audit` cluster setting]({% link {{ page.version.version
 
 Once `sql.log.user_audit` is set, the default behavior is for role-based SQL audit logging to take effect immediately within user sessions.
 
+`sql.log.user_audit.reduced_config.enabled`
+
+When enabled, the [`sql.log.user_audit.reduced_config.enabled` cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}#setting-sql-log-user-audit-reduced-config-enabled) computes a "reduced" audit configuration based on the user's current role memberships and the current value for the `sql.log.user_audit` cluster setting. The "reduced" audit configuration is computed at the **first SQL event emitted by the user, after the setting is enabled**. When the cluster setting is enabled, CockroachDB computes the audit configuration once at session start, instead of at each SQL event, thus providing around a 5% increase in throughput. However, changes to the audit configuration (user role memberships or cluster setting configuration) are not reflected within a session. To reflect the configuration changes in auditing behavior, users need to start a new session.
```
Comment (maryliag) [inline src/current/v23.1/role-based-audit-logging.md:56]: I don't think we should mention the percentage, other things could affect this value so is better to not add to the docs, so remove the sentence "thus providing around a 5% increase in throughput"

File: src/current/v23.2/role-based-audit-logging.md [hunk 2]
```diff
@@ -49,6 +51,10 @@ With the [`sql.log.user_audit` cluster setting]({% link {{ page.version.version
 
 Once `sql.log.user_audit` is set, the default behavior is for role-based SQL audit logging to take effect immediately within user sessions.
 
+`sql.log.user_audit.reduced_config.enabled`
+
+When enabled, the [`sql.log.user_audit.reduced_config.enabled` cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}#setting-sql-log-user-audit-reduced-config-enabled) computes a "reduced" audit configuration based on the user's current role memberships and the current value for the `sql.log.user_audit` cluster setting. The "reduced" audit configuration is computed at the **first SQL event emitted by the user, after the setting is enabled**. When the cluster setting is enabled, CockroachDB computes the audit configuration once at session start, instead of at each SQL event, thus providing around a 5% increase in throughput. However, changes to the audit configuration (user role memberships or cluster setting configuration) are not reflected within a session. To reflect the configuration changes in auditing behavior, users need to start a new session.
```
Comment (maryliag) [inline src/current/v23.2/role-based-audit-logging.md:56]: same comment as above

**Review by maryliag — CHANGES_REQUESTED** [review]

File: src/current/v23.1/role-based-audit-logging.md (hunk 1 above)
Comment (florence-crl) [reply src/current/v23.1/role-based-audit-logging.md]: removed

File: src/current/v23.2/role-based-audit-logging.md (hunk 2 above)
Comment (florence-crl) [reply src/current/v23.2/role-based-audit-logging.md]: removed

**Review by maryliag — APPROVED** [review]

File: src/current/v23.1/role-based-audit-logging.md [hunk 3]
```diff
@@ -49,6 +51,10 @@ With the [`sql.log.user_audit` cluster setting]({% link {{ page.version.version
 
 Once `sql.log.user_audit` is set, the default behavior is for role-based SQL audit logging to take effect immediately within user sessions.
 
+`sql.log.user_audit.reduced_config.enabled`
+
+When enabled, the [`sql.log.user_audit.reduced_config.enabled` cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}#setting-sql-log-user-audit-reduced-config-enabled) computes a "reduced" audit configuration based on the user's current role memberships and the current value for the `sql.log.user_audit` cluster setting. The "reduced" audit configuration is computed at the **first SQL event emitted by the user, after the setting is enabled**. When the cluster setting is enabled, CockroachDB computes the audit configuration once at session start, instead of at each SQL event. However with the setting enabled, changes to the audit configuration (user role memberships or cluster setting configuration) are not reflected within a user session. To reflect the configuration changes in auditing behavior, users need to start a new session.
```
Comment (rmloveland) [inline src/current/v23.1/role-based-audit-logging.md:56]: First mention of 'role memberships' could be a link to https://www.cockroachlabs.com/docs/stable/security-reference/authorization#roles 

First mention of 'user session' could be a link to https://www.cockroachlabs.com/docs/v23.1/show-sessions

File: src/current/v23.2/role-based-audit-logging.md [hunk 4]
```diff
@@ -49,6 +51,10 @@ With the [`sql.log.user_audit` cluster setting]({% link {{ page.version.version
 
 Once `sql.log.user_audit` is set, the default behavior is for role-based SQL audit logging to take effect immediately within user sessions.
 
+`sql.log.user_audit.reduced_config.enabled`
+
+When enabled, the [`sql.log.user_audit.reduced_config.enabled` cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}#setting-sql-log-user-audit-reduced-config-enabled) computes a "reduced" audit configuration based on the user's current role memberships and the current value for the `sql.log.user_audit` cluster setting. The "reduced" audit configuration is computed at the **first SQL event emitted by the user, after the setting is enabled**. When the cluster setting is enabled, CockroachDB computes the audit configuration once at session start, instead of at each SQL event. However with the setting enabled, changes to the audit configuration (user role memberships or cluster setting configuration) are not reflected within a user session. To reflect the configuration changes in auditing behavior, users need to start a new session.
```
Comment (rmloveland) [inline src/current/v23.2/role-based-audit-logging.md:56]: First mention of 'role memberships' could be a link to https://www.cockroachlabs.com/docs/stable/security-reference/authorization#roles

First mention of 'user session' could be a link to https://www.cockroachlabs.com/docs/v23.1/show-sessions

**Review by rmloveland — APPROVED** [review]
LGTM with some suggestions for more links

File: src/current/v23.1/role-based-audit-logging.md (hunk 3 above)
Comment (florence-crl) [reply src/current/v23.1/role-based-audit-logging.md]: added links

**Pull request merged**